bundle/edu.gemini.shared.util/src/main/java/edu/gemini/shared/util/immutable/ImEither.java

```diff
@@ -0,0 +1,105 @@
+package edu.gemini.shared.util.immutable;
+
+import java.io.Serializable;
+
+/**
+ * Simulates a Scalaz \/[S,T], and thus is similarly right-biased.
+ */
+public final class ImEither<L,R> implements Serializable {
+    private final Option<L> left;
+    private final Option<R> right;
+
```

Comment:
There is nothing in the types to prevent the implementation from constructing an `ImEither` that is both left and right, or neither. So I think something more like [this](https://github.com/functionaljava/functionaljava/blob/master/core/src/main/java/fj/data/Either.java) would be better. It's probably ok for now but please open an issue if it doesn't get fixed in this PR.